scripts/dts/gen_defines.py

```diff
@@ -298,20 +776,29 @@ def write_bus(node):
 def write_existence_flags(node):
     # Generate #defines of the form
     #
-    #   #define DT_INST_<instance no.>_<compatible string> 1
+    #   #define DT_<NODE_IDENT>      1
     #
     # for enabled nodes. These are flags for which devices exist.
+    #
+    # For example, for compat "vnd,device", this could look like:
+    #
+    #   #define DT_INST_0_VND_DEVICE 1
 
-    for compat in node.compats:
-        instance_no = node.edt.compat2enabled[compat].index(node)
-        out(f"INST_{instance_no}_{str2ident(compat)}", 1)
-
-
-def node_ident(node):
-    # Returns an identifier for 'node'. Used e.g. when building macro names.
-
-    # TODO: Handle PWM on STM
-    # TODO: Better document the rules of how we generate things
+    def write_existence_flags_for_ident(node, ident):
```

Comment:
I wondering if we should make this `DT_EXIST_<....> 1` or something.  As it stands the `#define DT_PATH_SOC_PINMUX_4004A000 1` ends up conflict with the prefix define for phandle(s).

```
/* phandle */
#define DT_PATH_SOC_GPIO_400FF080_NXP_KINETIS_PORT                      DT_PATH_SOC_PINMUX_4004A000
/* path exists */
#define DT_PATH_SOC_PINMUX_4004A000 1
```

Reply:
Sure, happy to tweak as needed. Is this a blocker for the other work you're doing? I can just restrict it to NODELABEL for now if that's unlikely to get us into trouble.

Reply:
Its a blocker so lets just restrict to previous defines and NODELABEL.